Feat/ai manager/match workers

Saikilito opened the pull request:
https://github.com/user-attachments/assets/e1031a6b-f4ab-400c-8d20-59e5b2188c09


https://github.com/user-attachments/assets/bc86499d-01ca-48b8-8e3c-8c9919980db3

Saikilito committed:
feat(migrations): add missing constraints and roles data seeding script

Saikilito committed:
fix(models): rename interface names for consistency in gig module

Saikilito committed:
feat(repositories): add drizzle-based repositories for gig-related models

Saikilito committed:
feat(repositories): update repository exports

Saikilito committed:
feat(review): add drizzle-based review repository and update exports

Saikilito committed:
fix(gig): update end_gig data type to boolean

Saikilito committed:
fix(review): update schema import to use reviews instead of recommendations

Saikilito committed:
feat(faker-api): add faker API with user data generation and removal

Saikilito committed:
fix(faker-api): enhance faker service and interfaces with new input types

Saikilito committed:
Merge pull request #50 from Saikilito/feat/add-fake-data-api

Feat/add fake data api

Saikilito committed:
Merge branch 'dev' of github.com:Saikilito/ableia into dev

Saikilito committed:
feat(database): add PostGIS extension and update related configurations

Saikilito committed:
refactor(ai-manager): fix assistant endpoint and related modules

Saikilito committed:
fix(models): update latitude and longitude types to number and add additionalInstructions to GigWork

Saikilito committed:
fix(faker-api): update latitude and longitude types to number in fakePrivateDataUser and generateFakeGigWork

Saikilito committed:
Merge pull request #51 from Saikilito/feat/GIG/add-postgis

feat(database): add PostGIS extension and update related configurations

Saikilito committed:
fix(maker): correct typo in message prompt and adjust loop condition for response handling

Saikilito committed:
fix(migrations): update latitude and longitude handling to ensure valid numeric conversion

Saikilito committed:
Merge branch 'dev' into fix/AI-MANAGER/fix-ai-endpoint

Saikilito committed:
fix(migrations): ensure proper type casting for latitude and longitude during update

Saikilito committed:
Merge pull request #52 from Saikilito/fix/AI-MANAGER/fix-ai-endpoint

Fix/ai manager/fix ai endpoint

Saikilito committed:
Merge branch 'dev' of github.com:Saikilito/ableia into dev

Saikilito committed:
fix(faker-api): standardize constant naming for badges, equipment, and skills

Saikilito committed:
fix(errors): correct error code for InternalServerError to match standard naming

Saikilito committed:
feat(repository): add findNearUsers method to PrivateDataUserRepository for proximity search

Saikilito committed:
feat(ai-manager): implement match workers use case and related interfaces

Saikilito committed:
feat(gemini): add match workers tool and related schemas

Saikilito committed:
feat(ai-manager-api): enhance database configuration and add match workers use case

Saikilito committed:
fix(ai-manager): improve match workers error handling

Saikilito committed:
Merge pull request #53 from Saikilito/feat/AI-MANAGER/match-workers

Feat/ai manager/match workers

Saikilito committed:
Merge pull request #10 from Israel-Laguan/feat/add-fake-data-api

Feat/add fake data api

Israel-Laguan commented on apps/ai-manager-api/src/dependency-injection/services/gemini.ts:
```diff
@@ -22,17 +29,29 @@ export const geminiService = {
+        responseSchema: {
+          type: SchemaType.OBJECT,
+          properties: {
+            text: {
```
Here add or just return a json with the array of candidates, or optionally error, can be something

Israel-Laguan commented on libs/product-domain/backend/src/ai-manager/infrastructure/gemini/client/maker.ts:
```diff
@@ -25,12 +22,43 @@ export function makeGeminiClient({ apiKey, llmConfig }: LlmGeminiServiceConfig)
     : [];
 
-  const newLlmConfig: ModelParams = {
+  const llmConfig_1: ModelParams = {
```
Why the change to snake case?

Saikilito committed:
feat(ai-manager): add retrieve buyer, user, and worker profile use cases and interfaces

Saikilito committed:
fix(gemini): rename llmConfig variables for consistency and improve error logging

Saikilito committed:
refactor(ai-manager): streamline match workers tool and integrate input schema

Saikilito committed:
feat(ai-manager): add gemini tools for retrieving buyer, user, and worker profiles

Saikilito committed:
feat(ai-manager-aí): integrate use cases for retrieving buyer, user, and worker profiles to assistant

Saikilito committed:
feat(ai-manager): update userId type to NUMBER in profile retrieval tools and add error handling for user statistics

Saikilito committed:
feat(models): add Location interface with latitude and longitude properties

Saikilito committed:
feat(ai-manager): enhance match workers functionality with location support and improve error handling

Saikilito committed:
feat(faker-api): implement generateFakeWorkers functionality and add input interface

Saikilito committed:
feat(faker-api): add POST request for generating fake workers with limit parameter

Saikilito committed:
fix(faker-api): refactor generateFakeWorkers for improved concurrency

Saikilito committed:
feat(migrations): update drizzle migration script

Saikilito committed:
fix(gig-db): improve exports structure

Saikilito committed:
feat(auth): implement switch user role use case and related interfaces

Saikilito committed:
feat(auth): add role repository

Saikilito committed:
feat(auth): add firebase switch user role service

Saikilito committed:
feat(auth-api): add switch user role endpoint

Saikilito committed:
Merge branch 'dev' of github.com:Saikilito/ableia into dev

Saikilito committed:
Merge pull request #54 from Saikilito/feat/AI-Manager/retrieve-profiles

Feat/ai manager/retrieve profiles

Saikilito committed:
Merge branch 'dev' of github.com:Saikilito/ableia into dev

Saikilito commented on apps/faker-api/src/api/faker/service.ts:
```diff
@@ -0,0 +1,289 @@
+
+  generateFakeBuyer: async (input: FakeBuyer) => {
+    const fakeBuyer: FakeBuyer = {
+      badgesAwarded: String(faker.helpers.arrayElements(FAKE_BADGES, { min: 1, max: 3 })),
```
Is not necesary

Saikilito committed:
feat(models): add match-workers and update-user interfaces; remove location interface

Saikilito committed:
feat(drizzle): add utility functions for SQL handling and constants

Saikilito committed:
refactor(auth): refactor user and privateDataUser repositories

Saikilito committed:
feat(worker): implement matchWorkers query

Saikilito committed:
fix(worker): fix matchWorkers zod schema

Saikilito committed:
refactor(match-workers): simplify matchWorkers use case and update interfaces

Saikilito committed:
refactor(ai-manager): update gemini MatchWorkersInputSchema

Saikilito committed:
feat(ai-manager-api): add match-worker endpoint

Saikilito committed:
Merge pull request #55 from Saikilito/feat/LIBS/refactor-drizzle-migration

feat(migrations): update drizzle migration script

Saikilito committed:
fix(match-workers): fix incorrect code

Saikilito committed:
feat(ai-manager-api): add authorization middleware

Saikilito committed:
fix(match-workers): update EQUIPMENT_MATCH_COUNT to use sql.raw

Saikilito committed:
fix(ai-manager-api): enhance match-workers request with additional parameters

Saikilito committed:
fix: update makeSQLArray to accept PostgreSQL type for safety queries

Saikilito committed:
fix(ai-manager-api): improve GOOGLE_SERVICE_ACCOUNT env validation and error handling

Saikilito committed:
fix:(libs): correct function name casing for makeFindUserIdsByPrivateDataUserIds

Saikilito committed:
Merge pull request #57 from Saikilito/refactor/match-workers

Refactor/match workers

Saikilito committed:
fix(ai-manager): remove unnecessary user profile statistic not found error check

Saikilito committed:
feat(ai-manager): add recommendation types and interfaces for recommendations and assistants

Saikilito committed:
refactor(ai-manager): enhance tool manager for multi input source support

Saikilito committed:
feat(ai-manager): add recommendations gemini output schema

Saikilito committed:
feat(ai-manager): add retrieve-raw-data gemini tool for data retrieval functionality

Saikilito committed:
feat(ai-manager): add recommendations endpoint and integrate hospitality skills taxonomy

Saikilito committed:
feat(faker-api): extend fake skills and bartender equipment for enhanced data generation

Saikilito committed:
feat(ai-manager): add role-based authorization middleware for enhanced security

Saikilito committed:
refactor(auth): rename MakeFirebaseSwitchUserRoleUseCase to MakeFirebaseSwitchUserRoleService for consistency

Saikilito committed:
fix(sql-migrations): add missing drop statements

Saikilito committed:
fix(match-workers): handle potential undefined statistic in matched workers

Saikilito committed:
Merge pull request #58 from Saikilito/refactor/AI-MANAGER/recomendation-endpoint

Refactor/ai manager/recomendation endpoint

Saikilito committed:
Merge branch 'dev' of github.com:Saikilito/ableia into dev

Saikilito committed:
Merge branch 'dev' into feat/AUTH/switch-role

Saikilito committed:
fix(auth): remove unused private data user repository implementation

Saikilito committed:
fix(match-workers): update skill matching to use ILIKE for case-insensitive search

Saikilito committed:
feat(faker-api): enhance fake data generation by using the project context

Saikilito committed:
feat(ai-manager): implement buyer and worker recommendation endpoints with updated service methods

Saikilito committed:
fix(service): correct worker recommendation function to use the appropriate assistant method
fix(match-workers): trim whitespace in skill matching for case-insensitive search

Saikilito committed:
Merge pull request #59 from Saikilito/fix/Ai-Manager

Fix/ai manager

Saikilito committed:
Merge branch 'dev' of github.com:Saikilito/ableia into dev

Saikilito committed:
Merge pull request #56 from Saikilito/feat/AUTH/switch-role

Feat/auth/switch role

Saikilito committed:
fix(gemini): update profile retrieval descriptions to invite users to create profiles if not found

Saikilito committed:
fix(ai-manager): rename llmConfig variables for clarity and consistency

Saikilito committed:
Merge pull request #60 from Saikilito/fix/Ai-Manager

Fix/ai manager

Saikilito committed:
Merge branch 'dev' of github.com:Saikilito/ableia into dev
